### PR TITLE
guide for cinn base operator contribute

### PR DESCRIPTION
**Pull request by thisjiang**
本PR为外部开发者提供待开发的CINN基础算子语义描述和开发指引。

### CODE REVIEW

File: pfcc/call-for-contributions/CINN_base_operator.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
+# CINN基础算子开发
+
+> This project will be mentored by [@zhhsplendid](https://github.com/zhhsplendid) and [@thisjiang](https://github.com/thisjiang)
+
+## 背景
+
+[CINN](https://github.com/PaddlePaddle/CINN)是一种在不改变源码的条件下加速Paddle模型运行速度的深度学习编译器。CINN致力于构造以训推一体自动调优、分布式为特色的科研界工业界亮点，吸引内外注意力影响力。
```
Comment (lanxianghit) [inline pfcc/call-for-contributions/CINN_base_operator.md:7]:
```suggestion
[CINN](https://github.com/PaddlePaddle/CINN)是一种在不改变模型代码的条件下加速飞桨模型运行速度的深度学习编译器。CINN致力于创造训推一体自动调优、分布式编译加速等特色功能，对深度学习模型提供全自动、极致的性能优化，并在科研界和工业界建立影响力。
```

File: pfcc/call-for-contributions/CINN_base_operator.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+不同于深度学习框架算子，在算子融合和自动调优技术的加持，深度学习编译器算子的粒度更细，算子数目也更少。在对接上层框架时，编译器会将上层的框架算子进一步拆分为若干基础算子，这样做的目的一方面是为了减少算子开发的工作量，通过仅实现有限的基础算子来组合出大量的上层框架算子；另一方面通过算子融合技术使得编译器可以跨算子融合，减少最终执行时的kernel数目，而自动调优技术使得编译器可以自动优化融合后的kernel，提升kernel性能。
```
Comment (lanxianghit) [inline pfcc/call-for-contributions/CINN_base_operator.md:9]:
```suggestion
不同于深度学习框架算子，深度学习编译器算子的粒度更细，算子数目也更少，因此在算子融合和自动调优方面具有更大的优势。在对接上层框架时，编译器会将上层的框架算子进一步拆分为若干基础算子，这样做的目的一方面是为了减少算子开发的工作量，仅实现有限的基础算子便可以组合出大量的上层框架算子；另一方面便于算子融合技术在编译器中可以实现跨算子自动融合，减少最终执行时的kernel数目和访存开销，达到更好的性能；此外，结合自动调优技术使得编译器可以自动优化融合后的kernel，提升kernel性能。
```

File: pfcc/call-for-contributions/CINN_base_operator.md (hunk 1 above)
Comment (thisjiang) [reply pfcc/call-for-contributions/CINN_base_operator.md]: Done

File: pfcc/call-for-contributions/CINN_base_operator.md (hunk 2 above)
Comment (thisjiang) [reply pfcc/call-for-contributions/CINN_base_operator.md]: Done

File: pfcc/call-for-contributions/CINN_base_operator.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+以`batch_norm + elementwise_add`算子为例，首先，`batch_norm`算子可以被拆分为32个基础算子，而后这些基础算子又可与`elementwise_add`相融合，最终可融合为两个kernel，测试结果表明，融合后的kernel在大部分配置下性能均能优于Paddle原生。
+
+![CINN overview](images/cinn_batchnorm_decomposer.png)
```
[the remark this answers is not in the document]
Comment (thisjiang) [reply pfcc/call-for-contributions/CINN_base_operator.md]: 已修改为`CINN BatchNorm Decomposer`

**Review by lanxianghit — DISMISSED** [review]
LGTM

**Review by wzzju — DISMISSED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
added index to this call for contribution to README.md
LGTM

**Pull request merged**